### ISSUE TITLE
Unable to use ingress resource and set routeAdvertisement for created virtualServer

### ISSUE THREAD

**Issue by ofen** [F5Networks/k8s-bigip-ctlr#1714]
Is there any way to specify routeAdvertisement for automatically created virtualServers/virtualAddresses via ingress resources?

It works if I use AS3 override via configMap like this:
```json
{
  "declaration": {
    "my_partition": {
      "defaultRouteDomain": 17,
      "Shared": {
        "anycast": {
          "class": "Service_Address",
          "virtualAddress": "10.0.0.12",
          "routeAdvertisement": "any"
        },
        "ingress_10_0_0_11_80": {
          "virtualAddresses": [
            {
              "use": "anycast"
            }
          ]
        }
      }
    }
  }
}
```

But I also want to keep old IP address from "virtual-server.f5.com/ip" ingress annotation (scheme will not allow to use "Service_Address" without "virtualAddress" and cannot be set to the same IP as in ingress annotation because of 422 response error from AS3).

**Comment by trinaths** [comment]
@ofen - What is the usecase you are trying to hold the old IP address ?

**Comment by ofen** [comment]
> @ofen - What is the usecase you are trying to hold the old IP address ?

Correct, we want to let users to allocation IP address via ingress resource (it would be better for the ingress controller to allocate IP by himself from specific IP range like ALB controller does but currently it's not supported yet).

**Comment by conkiztador** [comment]
Wondering if I'm seeing the same issue - currently when using Ingress definitions any virtual address is created with route advertisement disabled, so the IP gets no traffic. Is this intentional?

**Comment by mdditt2000** [comment]
routeAdvertisement was added to the CRD schema. Once i receive a test build from development i will pass it along. 

serviceAddress:
                  type: array
                  items:
                    type: object
                    properties:
                     arpEnabled:
                     type: boolean
                     icmpEcho:
                     type: string
                     **routeAdvertisement:
                     type: string**
                     spanningEnabled:
                     type: boolean
                     trafficGroup:
                     type: string
                     virtualAddress:
                     type: string

**Comment by ipanorik** [comment]
Guys, any updates on this problem?

**Comment by trinaths** [comment]
This issue is fixed in v2.4. routeAdvertisement is add to CRD Spec. https://github.com/F5Networks/k8s-bigip-ctlr/blob/master/docs/config_examples/crd/Install/customresourcedefinitions.yml#L63 
Closing this issue.
